src/Ilios/CoreBundle/DataFixtures/ORM/AbstractFixture.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace Ilios\CoreBundle\DataFixtures\ORM;
+
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Doctrine\Common\DataFixtures\AbstractFixture as DataFixture;
+use Doctrine\Common\DataFixtures\FixtureInterface;
+use Doctrine\Common\Persistence\ObjectManager;
+
+use Ilios\CoreBundle\Traits\IdentifiableEntityInterface;
+use Symfony\Component\HttpKernel\Config\FileLocator;
+use Symfony\Component\DependencyInjection\ContainerAwareInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * A Doctrine fixture based data loader base class that populates the DB from data files.
+ *
+ * Class AbstractFixture
+ * @package Ilios\CoreBundle\DataFixtures\ORM
+ */
+abstract class AbstractFixture extends DataFixture implements
+    FixtureInterface,
+    ContainerAwareInterface
+{
+    /**
+     * @var string
+     * Doubles as identifier for this fixture's data file and entity references.
+     */
+    protected $key;
+
+    /**
+     * @var boolean
+     * Set to TRUE if the loaded fixture should be held on for reference.
+     */
+    protected $storeReference;
+
+    /**
+     * @var ContainerInterface
+     */
+    private $container;
+
+
+
+    /**
+     * @param string $key
+     * @param boolean $storeReference
+     */
+    public function __construct($key, $storeReference = true)
+    {
+        $this->key = $key;
+        $this->storeReference = $storeReference;
+    }
+
+    public function getKey()
+    {
+        return $this->key;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setContainer(ContainerInterface $container = null)
+    {
+        $this->container = $container;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load(ObjectManager $manager)
+    {
+
+        $fileName = $this->getKey() . '.csv';
+        $path = $this->getDataFilePath($fileName);
+
+        $first = true;
+        if (($handle = fopen($path, 'r')) !== false) {
+            while (($data = fgetcsv($handle)) !== false) {
+                // step over the first row
+                // since it contains the field names
+                if ($first) {
+                    $first = false;
+                    continue;
+                }
+                $entity = $this->createEntity($data);
+                $manager->persist($entity);
+
+                // honor the given entity identifiers.
+                // @link http://www.ens.ro/2012/07/03/symfony2-doctrine-force-entity-id-on-persist/
+                $metadata = $manager->getClassMetaData(get_class($entity));
```

Comment:
There is a force ID option in the entity managers:
https://github.com/ilios/ilios/blob/master/src/Ilios/CoreBundle/Entity/Manager/AamcPcrsManager.php#L57

It may be too much work to inject our custom managers, but maybe that is what the $manager is.  I'm not sure.

I'm also worried that we're using a different pattern in these two places.  I have no idea what the correct path might be so I'm good with this one, just a comment.

Last, I think you can call these two metadata lines a single time before the loop as opposed to on each iteration.   There may be a slight performance bump in doing that, or maybe it will break everything.